scala.html.markdown

```diff
@@ -406,8 +407,63 @@ val otherGeorge = george.copy(phoneNumber = "9876")
 // And many others. Case classes also get pattern matching for free, see below.
 
 
-// Traits coming soon!
+// Traits
+Traits are somewhat similar to Java interfaces. With traits you define  
+a type and method signatures. Other than Java interfaces, Scala allows  
+you to mix method signatures with method implementations.  
+Parameters for constructors are not allowed.  
+Traits can inherit from other traits or classes, but only if they have  
+no parameters.  
+```
+trait Bird {
+	def breed: String
+	def color: String
+	def saySomething: String = "tweet"
+	def canFly: Boolean
+}
+class Duck extends Bird{
+	val breed = "bird"
+	val color = "green"
+	def canFly = true
+}
+```
+  
+```
+scala> var d = new Duck
+d: Duck = Duck@38c6f217  
+scala> d.breed
+res1: String = bird 
+scala> d.saySomething
+res0: String = tweet 
+scala> d.canFly
+res1: Boolean = true 
+```
 
+// Traits could be used via Mixins (Keyword "with")  
+```
+trait Sound {
+	def saySomething: String = "tweet tweet"
+}
+trait Bird {
```

Comment:
No need to define `Bird` again. Also please do not redefine `Duck`, that's the whole point of mixins, that you don't need to redefine the classes every time. In the illustration below, just say `d = new Duck with Sound`